node/src/effect.rs

```diff
@@ -964,6 +964,20 @@ impl<REv> EffectBuilder<REv> {
             .await
     }
 
+    /// Announce the intent to disconnect from a specific peer, which consensus thinks should be
+    /// punished.
```

Comment:
Or: "which consensus thinks is faulty"?
We're not punishing them, after all, apart from disconnecting.